specification/columns/availabilityzone.md

```diff
@@ -1,27 +1,27 @@
 # Availability Zone
 
-[Availability Zone](#glossary:availability-zone) is a provider assigned identifier for a physically separated and isolated area within a Region that provides high availability and fault tolerance. Availability Zone is commonly used for scenarios like analyzing cross-zone data transfer usage and the corresponding cost based on where [*resources*](#glossary:resource) are deployed.
+An [*availability zone*](#glossary:availability-zone) is a provider-assigned identifier for a physically separated and isolated area within a Region that provides high availability and fault tolerance. Availability Zone is commonly used for scenarios like analyzing cross-zone data transfer usage and the corresponding cost based on where [*resources*](#glossary:resource) are deployed.
 
-The AvailabilityZone column SHOULD be present in the billing data. This column MUST be of type String and MAY contain null values.
+The AvailabilityZone column MUST be present in the billing data when the provider supports deploying resources or services within an *availability zone*. This column MUST be of type String and MAY contain null values when a charge is not specific to an *availability zone*.
 
 ## Column ID
 
 AvailabilityZone
 
-## Display name
+## Display Name
 
 Availability Zone
 
 ## Description
 
-A provider assigned identifier for a physically separated and isolated area within a Region that provides high availability and fault tolerance.
+A provider-assigned identifier for a physically separated and isolated area within a Region that provides high availability and fault tolerance.
 
 ## Content constraints
 
 | Constraint      | Value            |
 |:----------------|:-----------------|
 | Column type     | Dimension        |
-| Column required | False            |
+| FOCUS Essential | False            |
```

Comment:
We introduce 'FOCUS Essential' in section 2 of the spec now (part of the PR). The hope is it becomes a easy marketing term that we can refer to for meeting a basic requirement set for FOCUS (without making a statement about anything beyond the essential columns - e.g. you can meet essential requirements and still fail FOCUS compliance). But yes, it will require some reading/thinking to know these are the must-have columns. If there is a different way to show this, I'm open to it.

For the options presented above: my challenge with anything that has 'required' is that in FOCUS Essential = false columns, the column can still be required IF the provider meets the criteria. E.g. Commitment commitment-based discount columns (e.g. CommitmentDiscountId) are not essential because many providers don't support commitment-based discounts. However, for providers that do support commitment-based discounts, it is required to be provided (see normative prose of the changes in this PR). So I worry about the term 'required' being used. Compliance level Essential can equal Full for providers that don't need to specify any columns outside of the essential. So that has some challenges too.

Once FinOps Foundation training and framework start referencing FOCUS, I think it won't be much of a stretch to think people in FinOps won't understand what 'FOCUS Essential' is without needing to reference section 2 of the spec.

Thoughts?

Reply:
I find FOCUS essential is nice to convey the message that this subset enables essential FinOps activities, and that's the minimal subset a provider must support to claim to support FOCUS.

Furthermore, if a provider supports a non-essential column, is there a known or foreseen case where the provider would make the column absent in the FOCUS dataset? And if yes, would it be an issue? If yes, would stating the dataset is not FOCUS compliant help to solve this issue?

Reply:
> my challenge with anything that has 'required' is that in FOCUS Essential = false columns, the column can still be required IF the provider meets the criteria.

In this case, "Required: When supported" would be more explicit than "FOCUS Essential: False".

The idea of "Compliance level: Essential" was trying to keep the marketing aspect of "Essential" while accepting that we'll likely need multiple compliance levels. That makes "Essential" a lot more acceptable as added complexity since it's one of the several levels of compliance.

That said, "Essential" is not a compliance level based on this description. A compliance level is something you can support without doing anything more. So regardless of whether I support commitment discounts or not, I could declare Essential compliance without those columns. However, if we say you MUST add those columns, then that means it's never possible to be Essential-compliant without being "fully-compliant" (accepting that full compliance allows for not including unsupported columns).

Unless I'm missing something, this minimizes the value of "Essential" as a marketing term since it's not a statement of compliance. It just conveys to practitioners which columns will be in every FOCUS dataset but it won't be useful when determining what a provider supports. From that perspective, everyone who supports FOCUS would be "Essential".

It might be useful to look at how others use compliance levels. WCAG (accessibility) is my go-to. Sites can either be A, AA, or AAA based on meeting specific guidelines. This is easy for website owners to abide by and absolutely critical for end users who might rely on something that's only supported in AAA. For them, they know that they cannot use a product unless it's AAA. That's immensely powerful. If we follow this model, we might have compliance levels like:
1. `Basic` = "Essential" (or "always" required) columns only, regardless of what's supported. (Alt: Core, Essential)
2. `Standard` = Core + all columns that map to features supported by the provider (e.g., AZ, CBD). (Alt: Feaure)
3. `Extended` = Every column, regardless of what's supported. (Alt: Full, Complete)
4. `Complete` = Every column, regardless of what's supported, with all MUST and SHOULD rules applied. (Alt: Comprehensive)

🤔

Reply:
FOCUS Essential is not a compliance level. It's a way to indicate that a minimum set of important columns are available in the dataset. 

Compliance levels should be handled post 1.0. Many good patterns as you've all called out. To me, it's orthogonal to this discussion. Today, you can have essential columns and columns that are required for subsets of providers based on their capabilities.